thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java

```diff
@@ -249,6 +256,33 @@ public void onChange(final TiPresenter.State state,
     }
 
     @Test
+    public void testRemoveOtherObserver() throws Exception {
+        mPresenter.create();
+
+        final TiLifecycleObserver observer1 = mock(TiLifecycleObserver.class);
+        mPresenter.addLifecycleObserver(observer1);
+        final TiLifecycleObserver observer2 = mock(TiLifecycleObserver.class);
+        final Removable removable = mPresenter.addLifecycleObserver(observer2);
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) throws Throwable {
+                removable.remove();
+                return null;
+            }
+        }).when(observer1).onChange(any(TiPresenter.State.class), anyBoolean());
+
+        mPresenter.attachView(mock(TiView.class));
+
+        //receives both events
+        verify(observer1, times(2)).onChange(any(TiPresenter.State.class), anyBoolean());
```

Comment:
For understadning:
We call `attachView` which triggers the `fori`.
The list have two items. Observable1 and Observable2.
`observable1#onChange` got called with `init` and `attach`. Which triggers the answer to remove the `observable2`from the Observable list.
But because of the fact we have a local copy, `observable2#onChange` got also called with `init`. But not with `attach` anymore...

Right? 🤔